new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java

```diff
@@ -0,0 +1,312 @@
+
+    // wait for the domain to exist
+    logger.info("Check for domain custom resource in namespace {0}", domainNamespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for domain {0} to be created in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                domainUid,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainExists(domainUid, DOMAIN_VERSION, domainNamespace));
+  }
+
+  /**
+   * Verify all server pods are running.
+   * Verify all k8s services for all servers are created.
+   */
+  @BeforeEach
+  public void beforeEach() {
+    logger.info("Check admin service and pod {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace);
+    // check managed server services and pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server services and pods are created in namespace {0}",
+          domainNamespace);
+      checkPodReadyAndServiceExists(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+  }
+
+  /**
+   * Create a WebLogic domain with `AdministrationPortEnabled: true`.
+   * Create a domain CRD with a channel with the name `default-admin`.
+   * Make sure an external NodePort service is created in domain namespace.
+   * Make sure WebLogic console is accessible through the `default-admin` service.  
+   * Make sure WebLogic console is not accessible through the `default` service.  
+   */
+  @Test
+  @DisplayName("Verify the secure service through administration port")
+  public void testVerifyAdminPortService() {
+    int sslNodePort = getServiceNodePort(
+         domainNamespace, adminServerPodName + "-external", "default-admin");
+    logger.info("Found the administration service nodePort {0}", sslNodePort);
+    String curlCmd = "curl -sk --show-error --noproxy '*' "
+        + " https://" + K8S_NODEPORT_HOST + ":" + sslNodePort
+        + "/console/login/LoginForm.jsp --write-out %{http_code} -o /dev/null";
+    logger.info("Executing defaukt-admin nodeport curl command {0}", curlCmd);
+    assertTrue(callWebAppAndWaitTillReady(curlCmd, 10));
+    logger.info("WebLogic console is accessible thru default-admin service");
+
+    int nodePort = getServiceNodePort(
+           domainNamespace, adminServerPodName + "-external", "default");
+    logger.info("Found the default service nodePort {0}", nodePort);
+    curlCmd = "curl -s --show-error --noproxy '*' "
+        + " http://" + K8S_NODEPORT_HOST + ":" + nodePort
+        + "/console/login/LoginForm.jsp --write-out %{http_code} -o /dev/null";
+    logger.info("Executing default nodeport curl command {0}", curlCmd);
+    assertFalse(callWebAppAndWaitTillReady(curlCmd, 5));
```

Comment:
same as above comment

Reply:
Now I have added a check for the NodePort Service to make sure that, it finds a valid integer port.

int nodePort = getServiceNodePort(
       domainNamespace, adminServerPodName + "-external", "default");
assertTrue(nodePort != -1 ,  "Could not get the default external service node port");